app/src/main/java/org/mozilla/fenix/perf/StartupManager.kt

```diff
@@ -0,0 +1,280 @@
+        @Suppress("LongParameterList", "LongMethod")
+        fun activityOnCreate(
+            activity: HomeActivity,
+            savedInstanceState: Bundle?,
+            isVisuallyComplete: Boolean,
+            webExtensionPopupFeature: WebExtensionPopupFeature,
+            getBreadcrumbMessage: (NavDestination) -> String,
+            externalSourceIntentProcessors: KProperty0<List<HomeIntentProcessor>>,
+            action: () -> Unit
+        ): StartupData {
+            var themeManager: ThemeManager?
+            var browsingModeManager: BrowsingModeManager?
+
+            activity.components.strictMode.attachListenerToDisablePenaltyDeath(activity.supportFragmentManager)
+            // There is disk read violations on some devices such as samsung and pixel for android 9/10
+            action()
+
+            // Diagnostic breadcrumb for "Display already aquired" crash:
+            // https://github.com/mozilla-mobile/android-components/issues/7960
+            activity.breadcrumb(
+                message = "onCreate()",
+                data = mapOf(
+                    "recreated" to (savedInstanceState != null).toString(),
+                    "intent" to (activity.intent?.action ?: "null")
+                )
+            )
+
+            activity.components.publicSuffixList.prefetch()
+
+            val pair = setupThemeAndBrowsingMode(getModeFromIntentOrLastKnown(activity.intent, activity), activity)
+            themeManager = pair.first
+            browsingModeManager = pair.second
+
+            browsingModeManager.addListener { newMode ->
+                themeManager.currentTheme = newMode
+            }
+
+            activity.setContentView(R.layout.activity_home)
+
+            val navHost = activity.supportFragmentManager
+                                        .findFragmentById(R.id.container) as NavHostFragment
+            // Must be after we set the content view
+            if (isVisuallyComplete) {
+                activity.components.performance.visualCompletenessQueue
+                    .attachViewToRunVisualCompletenessQueueLater(
+                        WeakReference(activity.findViewById<LinearLayout>(R.id.rootContainer))
+                    )
+            }
+
+            val sessionObserver = UriOpenedObserver(activity)
+
+            checkPrivateShortcutEntryPoint(activity.intent)
+            val privateNotificationObserver = PrivateNotificationFeature(
+                activity.applicationContext,
+                activity.components.core.store,
+                PrivateNotificationService::class
+            ).also {
+                it.start()
+            }
+
+            if (isActivityColdStarted(activity.intent, savedInstanceState)) {
+                externalSourceIntentProcessors.get().any {
+                    it.process(
+                        activity.intent,
+                        navHost.navController,
+                        activity.intent
```

Comment:
Why is intent passed twice? 🤔

Reply:
Good question. Looking at the method from `HomeIntentProcessor.process` it seems as if the first param is: "The intent to process" and the third param ( Or second intent) is " intent to mutate"

Reply:
Weird. I wonder if we should have someone from android components look this over just to make sure it's correct because it seems strange to me.